legend-sdlc-server-shared/src/main/java/org/finos/legend/sdlc/server/BaseServer.java

```diff
@@ -98,7 +95,7 @@ public void run(C configuration, Environment environment)
         corsFilter.setInitParameter(CrossOriginFilter.ALLOWED_METHODS_PARAM, "GET,PUT,POST,DELETE,OPTIONS");
         corsFilter.setInitParameter(CrossOriginFilter.ALLOWED_ORIGINS_PARAM, "*");
         corsFilter.setInitParameter(CrossOriginFilter.ALLOWED_TIMING_ORIGINS_PARAM, "*");
-        corsFilter.setInitParameter(CrossOriginFilter.ALLOWED_HEADERS_PARAM, "X-Requested-With,Content-Type,Accept,Origin,Access-Control-Allow-Credentials,x-b3-parentspanid,x-b3-sampled,x-b3-spanid,x-b3-traceid");
+        corsFilter.setInitParameter(CrossOriginFilter.ALLOWED_HEADERS_PARAM, "X-Requested-With,Content-Type,Accept,Origin,Access-Control-Allow-Credentials" + (configuration.getCORSConfiguration() != null && configuration.getCORSConfiguration().getAllowedHeaders() != null ? LazyIterate.adapt(configuration.getCORSConfiguration().getAllowedHeaders()).makeString(",", ",", "") : ""));
```

Comment:
I think what we should do here is allow users to set the full list of allowed headers. If the config is absent, the default value should be the previous set. That way we will maximize configurability while retaining the previous behavior when the config is absent.

Reply:
Makes sense. I have addressed this. Also, interestingly, I found that actually non of those headers are **essential**, by that I mean I try to run my local without any of things still work just fine. I left a comment there so we could be reminded that they are left for backward compatibility and we should consider removing them eventually.